domain/src/DomainStorage.php

```diff
@@ -98,7 +98,7 @@ public function loadDefaultDomain() {
    * @return \Drupal\domain\DomainInterface[]
    */
   public function loadMultiple(array $ids = NULL) {
-    $domains = $this->loadMultiple($ids);
+    $domains = parent::loadMultiple($ids);
```

Comment:
If we're calling parent here, do we even need this method override? 

This may be a legacy method from before EntityStorage was standardized.

Reply:
This and some others are my fault: I wanted the functions in DomainStorage class to be declared as returning items of type DomainInterface, so the type checking in phpcs correctly reports on functions and their args.

I am very happy to remove the additional code if there is a better way. Might that way be using @method tags: http://docs.phpdoc.org/references/phpdoc/tags/method.html / https://www.drupal.org/project/coding_standards/issues/2920333 ?

If so the whole loadMultiple function and docblock referenced here could, I think, be reduced to:

    /**
     * @method \Drupal\domain\DomainInterface[] loadMultiple(array $ids = NULL)
     */

Reply:
The use of parent here is a code smell.

I wouldn't rewrite that, because it confuses developers used to the core API. Returning generic entities are fine. I think you can still type hint on the array elements returned by EntityStorageBase::loadMultiple() if you wanted to.

The `sort` method called by LoadMultipleSorted() does that.

```
  /**
   * {@inheritdoc}
   */
  public function sort(DomainInterface $a, DomainInterface $b) {
    return $a->getWeight() > $b->getWeight();
  }
```

Reply:
I agree about the smell, but the case of sort() here is for parameters, not function results.

Just to be sure, did you realize that I added the whole loadMultiple() function around line 100, not just changed line 101?

I don't really understand "confuses developers used to the core API", given that these functions are being called on domain objects and have the same function signature and results as for the entity api: could you elaborate?

I wonder if it would be useful to meet on drupal slack tomorrow to clarify this and any other issues?

Reply:
Sure. I can be in Slack. I'm around today, too.

I only noticed you added this method after the initial reply. 

The sort method definition type hints the array elements returned by loadMultiple(), knowing that the base class will return DomainInterface objects as an array.

```
 public function sort(DomainInterface $a, DomainInterface $b)
```

The confusion is that our class has a method that overwrites the base class but merely invokes it as `parent`, which makes me scratch my head as a developer. Why override the base method if it isn't changed?